modules/rotating-bound-service-keys.adoc

```diff
@@ -0,0 +1,404 @@
+  --key keys.json ${TEMPDIR}/jwks.current.json
+----
+endif::aws[]
+ifdef::gcp[]
++
+[source,terminal]
+----
+$ gcloud storage cp gs://${GCP_BUCKET}/keys.json ${TEMPDIR}/jwks.current.json
+----
+endif::gcp[]
+ifdef::azure[]
++
+[source,terminal]
+----
+$ az storage blob download \
+  --container-name ${AZURE_STORAGE_CONTAINER} \
+  --account-name ${AZURE_STORAGE_ACCOUNT} \
+  --name 'openid/v1/jwks' \
+  -f ${TEMPDIR}/jwks.current.json
+----
+endif::azure[]
+
+. Combine the two `keys.json` files by running the following command:
++
+[source,terminal]
+----
+$ jq -s '{ keys: map(.keys[])}' ${TEMPDIR}/jwks.current.json ${TEMPDIR}/jwks.new.json > ${TEMPDIR}/jwks.combined.json
+----
+
+. To enable authentication for the old and new keys during the rotation, upload the combined `keys.json` file to the cloud provider by running the following command:
+ifdef::aws[]
++
+[source,terminal]
+----
+$ aws s3api put-object \
+  --bucket ${AWS_BUCKET} \
+  --tagging "openshift.io/cloud-credential-operator/${CLUSTER_NAME}=owned" \
+  --key keys.json \
+  --body ${TEMPDIR}/jwks.combined.json
+----
+endif::aws[]
+ifdef::gcp[]
++
+[source,terminal]
+----
+$ gcloud storage cp ${TEMPDIR}/jwks.combined.json gs://${GCP_BUCKET}/keys.json
+----
+endif::gcp[]
+ifdef::azure[]
++
+[source,terminal]
+----
+$ az storage blob upload \
+  --overwrite \
+  --account-name ${AZURE_STORAGE_ACCOUNT} \
+  --container-name ${AZURE_STORAGE_CONTAINER} \
+  --name 'openid/v1/jwks' \
+  -f ${TEMPDIR}/jwks.combined.json
+----
+endif::azure[]
+
+. Wait for the Kubernetes API server to update and use the new key. 
+You can monitor the update progress by running the following command:
++
+[source,terminal]
+----
+$ oc adm wait-for-stable-cluster
+----
++
+This process might take 15 minutes or longer.
+The following output indicates that the process is complete:
++
+[source,text]
+----
+All clusteroperators are stable
+----
+
+. To ensure that all pods on the cluster use the new key, you must restart them.
++
+[IMPORTANT]
+====
+This step maintains uptime for services that are configured for high availability across multiple nodes, but might cause downtime for any services that are not.
+====
++
+Restart all of the pods in the cluster by running the following command:
++
+[source,terminal]
+----
+$ oc adm reboot-machine-config-pool mcp/worker mcp/master
+----
+
+. Monitor the restart and update process by running the following command:
++
+[source,terminal]
+----
+$ oc adm wait-for-node-reboot nodes --all
+----
++
+This process might take 15 minutes or longer.
+The following output indicates that the process is complete:
++
+[source,text]
+----
+All nodes rebooted
+----
+
+. Monitor the update progress by running the following command:
++
+[source,terminal]
+----
+$ oc adm wait-for-stable-cluster
+----
++
+This process might take 15 minutes or longer.
+The following output indicates that the process is complete:
++
+[source,text]
+----
+All clusteroperators are stable
+----
```

Comment:
[These steps in the dev repo doc](https://github.com/openshift/cloud-credential-operator/blob/6e5ca5b2fe113771c395210ae11dee756e074152/docs/rotate-oidc-key.md?plain=1#L145) were familiar from my work in [Enabling Microsoft Entra Workload ID on an existing cluster](https://docs.openshift.com/container-platform/4.17/post_installation_configuration/changing-cloud-credentials-configuration.html#enabling-entra-workload-id-existing-cluster_changing-cloud-credentials-configuration), so I copied them over. Let me know if that is appropriate here.

Reply:
Looks good.